homeassistant/components/google/config_flow.py

```diff
@@ -118,10 +122,10 @@ async def async_step_auth(
                 )
                 return self.async_abort(reason="oauth_error")
             calendar_access = get_feature_access(self.hass)
-            if self._reauth_config_entry and self._reauth_config_entry.options:
-                calendar_access = FeatureAccess[
-                    self._reauth_config_entry.options[CONF_CALENDAR_ACCESS]
-                ]
+            if self.source == SOURCE_REAUTH and (
+                reauth_options := self._get_reauth_entry().options
+            ):
+                calendar_access = FeatureAccess[reauth_options[CONF_CALENDAR_ACCESS]]
```

Comment:
Note: coverage is already missing in `dev` - not linked to this PR.

Reply:
Can we add coverage?

Reply:
> Can we add coverage?

@allenporter can you help?

Reply:
Yes https://github.com/home-assistant/core/pull/128690